File: src/ClientApp/src/app/pages/clients/view-models/client-product-lookup-bundle.view-model.ts
import { LookupViewModel } from '../../leads/view-models/lookup.view-model';
import { ClientUserLookupViewModel } from './client-user-lookup.view-model';

export interface ClientOpportunityLookupViewModel {
    id: string;
    clientId: string;
    productId: string;
    opportunityNumber: string;
    title: string;
    displayName: string;
}

export interface ClientProductLookupBundleViewModel {
    products: LookupViewModel[];
    owners: ClientUserLookupViewModel[];
    opportunities: ClientOpportunityLookupViewModel[];
    relationshipStatuses: LookupViewModel[];
}

File: src/API/src/Modules/Clients/Clients.Application/ViewModels/ClientProductLookupBundleViewModel.cs
namespace Clients.Application.ViewModels
{
    public class ClientProductLookupBundleViewModel
    {
        public List<LookupViewModel> Products { get; set; } = new();
        public List<ClientUserLookupViewModel> Owners { get; set; } = new();
        public List<ClientOpportunityLookupViewModel> Opportunities { get; set; } = new();
        public List<LookupViewModel> RelationshipStatuses { get; set; } = new();
    }

    public class ClientOpportunityLookupViewModel
    {
        public Guid Id { get; set; }
        public Guid ClientId { get; set; }
        public Guid ProductId { get; set; }
        public string OpportunityNumber { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string DisplayName => $"{OpportunityNumber} - {Title}";
    }
}
